From 69adf3283e18724f484e06bedd7ef299d6f99821 Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Mon, 27 Oct 2014 18:36:34 -0400
Subject: [x86] quirks: use gen6 stolen detection for VLV

Message-id: <1414434997-7515-4-git-send-email-rclark@redhat.com>
Patchwork-id: 98476
O-Subject: [RHEL7 drm 3/6] x86/early quirk: use gen6 stolen detection for VLV
Bugzilla: 1154053
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

RHBZ: 1154053
Upstream: 7bd40c16ccb2cb6877dd00b0e66249c171e6fa43

commit 7bd40c16ccb2cb6877dd00b0e66249c171e6fa43
Author:     Jesse Barnes <jbarnes@virtuousgeek.org>
AuthorDate: Tue Nov 12 10:17:39 2013 -0800
Commit:     Daniel Vetter <daniel.vetter@ffwll.ch>
CommitDate: Thu Nov 14 09:32:11 2013 +0100

    x86/early quirk: use gen6 stolen detection for VLV

    We've always been able to use either method on VLV, but it appears more
    recent BIOSes only support the gen6 method, so switch over to that.

    References: https://bugs.freedesktop.org/show_bug.cgi?id=71370
    Signed-off-by: Jesse Barnes <jbarnes@virtuousgeek.org>
    Reviewed-by: Ville Syrjala <ville.syrjala@linux.intel.com>
    Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>

Signed-off-by: Rob Clark <rclark@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kernel/early-quirks.c b/arch/x86/kernel/early-quirks.c
index 96f958d..bc4a088 100644
--- a/arch/x86/kernel/early-quirks.c
+++ b/arch/x86/kernel/early-quirks.c
@@ -330,8 +330,8 @@ static struct pci_device_id intel_stolen_ids[] __initdata = {
  INTEL_I915GM_IDS(gen3_stolen_size),
  INTEL_I945G_IDS(gen3_stolen_size),
  INTEL_I945GM_IDS(gen3_stolen_size),
- INTEL_VLV_M_IDS(gen3_stolen_size),
- INTEL_VLV_D_IDS(gen3_stolen_size),
+ INTEL_VLV_M_IDS(gen6_stolen_size),
+ INTEL_VLV_D_IDS(gen6_stolen_size),
  INTEL_PINEVIEW_IDS(gen3_stolen_size),
  INTEL_I965G_IDS(gen3_stolen_size),
  INTEL_G33_IDS(gen3_stolen_size),
-- 
1.7.1